helm-chart/binderhub/templates/configmap.yaml

```diff
@@ -3,19 +3,24 @@ apiVersion: v1
 metadata:
   name: binder-config
 data:
-{{- /* Important: only pick relevant subset of values. Including too much here can trigger unnecessary binderhub pod restarts */}}
-{{- $values := pick .Values "config" "cors" "dind" "extraConfig" }}
-{{- /* make a copy to avoid modifying the original */ -}}
-{{- $_ := set $values "config" (merge dict .Values.config) }}
-{{- $_ := set $values.config "BinderHub" (merge dict .Values.config.BinderHub) }}
-{{- /* trim secret values. Update here if new secrets are added! */ -}}
-{{- /* every 'omit' here should be matched with a corresponding 'pick' in secret.yaml */ -}}
-{{- if $values.config.GitHubRepoProvider }}
-{{- $_ := set $values.config "GitHubRepoProvider" (omit .Values.config.GitHubRepoProvider "client_id" "client_secret" "access_token") }}
-{{- end }}
-{{- if $values.config.GitLabRepoProvider }}
-{{- $_ := set $values.config "GitLabRepoProvider" (omit .Values.config.GitLabRepoProvider "private_token" "access_token") }}
-{{- end }}
+  {{- /*
+  FIXME: Stop splitting up parts into configmap/secret as it adds complexity,
+         instead only use a secret. For the equivalent PR in z2jh, see:
+         https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1682
+  */}}
```

Comment:
Why add comments like this into the code instead of making an issue? I think we should try and keep issues as "the one place to go" to find out about things that could be done. Even if only because it is much easier to clean up and close stale issues than clean up and remove stale comments like this.

Reply:
While browsing through the files I realized this, and knew that I could create an issue, but I also felt like advertising this within this context so that everyone working in this file get a notice and encouragement to fix this.

Reply:
Created https://github.com/jupyterhub/binderhub/issues/1176 to represent this as well.